sdk/tables/Azure.Data.Tables/src/DynamicTableEntity.cs

```diff
@@ -0,0 +1,226 @@
+        /// meaning that each time the entity is modified, the value of Timestamp increases for that entity. This property should not be set on insert or update operations (the value will be ignored).
+        /// </summary>
+        /// <value>A <see cref="DateTimeOffset"/> containing the timestamp of the entity.</value>
+        public DateTimeOffset Timestamp
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.TimeStamp, out object value) ? (DateTimeOffset)value : default; }
+        }
+
+        /// <summary>
+        /// Gets or sets the entity's ETag. Set this value to '*' in order to force an overwrite to an entity as part of an update operation.
+        /// </summary>
+        /// <value>A string containing the ETag value for the entity.</value>
+        public string ETag
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.Etag, out object value) ? (string)value : default; }
+            set { _properties[TableConstants.PropertyNames.Etag] = value; }
+        }
+
+        /// <summary>
+        /// Constructs an instance of a <see cref="TableEntity" />.
+        /// </summary>
+        public DynamicTableEntity()
+            : this(null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DynamicTableEntity"/> class with the specified partition key and row key.
+        /// </summary>
+        /// <param name="partitionKey">A string containing the partition key of the <see cref="DynamicTableEntity"/> to be initialized.</param>
+        /// <param name="rowKey">A string containing the row key of the <see cref="DynamicTableEntity"/> to be initialized.</param>
+        public DynamicTableEntity(string partitionKey, string rowKey)
+            : this(null)
+        {
+            PartitionKey = partitionKey;
+            RowKey = rowKey;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DynamicTableEntity"/> class with properties in the <see cref="IDictionary"/>.
+        /// </summary>
+        /// <param name="values">A <see cref="IDictionary"/> containing the initial values of the entity.</param>
+        public DynamicTableEntity(IDictionary<string, object> values)
+        {
+            _properties = values != null ?
+                new Dictionary<string, object>(values) :
+                new Dictionary<string, object>();
+        }
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="String"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public string GetString(string key) => GetValue<string>(key);
```

Comment:
We should document that all of these properties can throw `InvalidOperationException` in the event of a type mismatch.

ex:
```c#
/// <exception cref="InvalidOperationException"><paramref name="key"/> is not of type <see cref="string" />.</exception>
/// <exception cref="KeyNotFoundException"><paramref name="key"/> is not not found in the dictionary.</exception>
```

Reply:
[tracking here](https://github.com/Azure/azure-sdk-for-net/issues/13658)